### PR TITLE
stop ignoring -Wsign-compare globally and incrementally fix

### PR DESCRIPTION
**Pull request by dagar**
More incremental approach to https://github.com/PX4/Firmware/pull/7346.

### CODE REVIEW

File: src/drivers/iridiumsbd/IridiumSBD.cpp [hunk 1]
```diff
@@ -275,7 +275,7 @@ void IridiumSBD::standby_loop(void)
 	// check for incoming SBDRING, handled inside read_at_command()
 	read_at_command();
 
-	if (param_read_interval_s != 0 && (hrt_absolute_time() - last_read_time) / 1000000 > param_read_interval_s) {
+	if (param_read_interval_s != 0 && ((hrt_absolute_time() - last_read_time) > param_read_interval_s * 1e6)) {
```
Comment (LorenzMeier) [inline src/drivers/iridiumsbd/IridiumSBD.cpp]: 1e6 is a double - not the right constant here.

Comment (dagar) [inline src/drivers/iridiumsbd/IridiumSBD.cpp]: Fixed.

**Comment by TSC21** [comment]
@dagar are you able to fix the problems reported by the CI system?

**Comment by dagar** [comment]
Rebased

**Comment by dagar** [comment]
Jenkins build failure seems unrelated.

**Pull request merged**